### ISSUE TITLE
False positive validation

### ISSUE THREAD

**Issue by kriptonuz** [yiisoft/validator#118]
Number rule gives false positive on boolean values
### What steps will reproduce the problem?
```
$rule = new \Yiisoft\Validator\Rule\Number();
$result = $rule->validate(false);
var_dump($result->isValid());
$result = $rule->validate(true);
var_dump($result->isValid());
```
### What is the expected result?
```
false
false
```
### What do you get instead?
```
true
true
```
### Additional info

| Q                | A
| ---------------- | ---
| Version          | 1.0.?
| PHP version      | N/A
| Operating system | N/A

**Comment by rustamwin** [comment]
Boolean values will be converted to integers during validation.

**Comment by kriptonuz** [comment]
Is it normal? For example if I write rule (new Number())->integer(), I will expect integer not boolean values

**Comment by samdark** [comment]
That is tricky. If we deal with standard browser form requests, everything comes as a string from these. JSON-bodies are a bit better but still not ideal. Therefore, type-casting makes sense for input. From another standpoint, using validators standalone is different and sticter comparison is expected.

**Comment by kriptonuz** [comment]
Now I'm using yiisoft/validator for REST API, and I'm validating data came as Json. There are several use cases like this

**Comment by samdark** [comment]
Yes, definitely. Need to think about it.

**Comment by viktorprogger** [comment]
What about implementing a `strictTypes` configuration parameter for the rule?

**Comment by samdark** [comment]
Yeah. That may help. Parameter could be just `strict()`.
